src/quo/components/wallet/address_text/view.cljs

```diff
@@ -0,0 +1,26 @@
+(ns quo.components.wallet.address-text.view
+  (:require [quo.components.markdown.text :as text]
+            [quo.foundations.colors :as colors]
+            [utils.address :as utils]))
+
+(defn- view-internal
+  [{:keys [networks address type blur? theme]}]
+  [text/text
+   (map (fn [network]
+          ^{:key (str network)}
+          [text/text
+           {:size  :paragraph-2
+            :style {:color (colors/resolve-color network theme)}}
+           (str (subs (name network) 0 3) (when (= network :arbitrum) "1") ":")])
```

Comment:
For this piece of code, I'd prefer to generate all the hiccup instead of having a lazy seq.

The reason is that with the lazy sequence, we have to set IDs, in this implementation we are setting `network` as ID, which is highly possible to be repeated when used in the same screen along with other components (that'll be bad for react).

To do this, we can use `(into [text/text] my-hiccup)`

And about the way we are extracting the short names, a more general solution would be to have a map that takes network names (e.g. :ethereum) and returns short names (e.g. "eth"), and extract it with a function that also has a fallback option.

Remember that we have styles for an unknown network, so when someone provides `:my-network`, I think we should use the unknown color and "my-network:" as part of the address string.

BTW I already handle this in the `share-qr-code` multichain variant I'm currently working on, so this PR will affect what I'm doing too 👀

Reply:
Sure, I mean at the very least this should be a utility function if we're handling the same logic in multiple places. 
If some details change about this then we have to adjust this in multiple places.
Additionally if a new network s added and it needs some special handling similar to 'arb' then we have to update all uses to account for this too.
Given that the network names are known I don't think we need a fallback tbh but sure we can add it 👍

Reply:
Good point about the keys @ulisesmac

Reply:
Hey @OmarBasem, in my PR I was suggesting avoid using `^:key` for this text generation, related to only that I said you could use `into`. Here's a code thaqt might be used, I split it into multiple functions trying to improve the readability, but feel free to change everything you want :) just a suggestion:

```clojure

;; BTW, network colors don't need the `theme` prop since they don't have theme variants,
;; but IDK, maybe it's ok to pass the prop if we have it available
(defn- colored-network-text [theme network]
  [text/text {:size  :paragraph-2
              :style {:color (colors/resolve-color network theme)}}
   ;; added some `\n` between the params of `str` so now it's easier to read what
   ;; parameters is str concatenating
   (str (subs (name network) 0 3)
        (when (= network :arbitrum) "1")
        ":")])

(defn- address-text-style [address-type blur? theme]
  (when (= address-type :short)
    {:color (if blur?
              colors/white-opa-40
              (colors/theme-colors colors/neutral-50 colors/neutral-40 theme))}))

(defn- view-internal
  [{:keys        [networks address blur? theme]
    address-type :type}] ;; Renamed to avoid shadowing cljs.core/type
  (let [network-text-xf (map #(colored-network-text theme %))
        address-text    [text/text
                         {:size   :paragraph-2
                          ;; TODO: monospace font https://github.com/status-im/status-mobile/issues/17009
                          :weight :monospace
                          :style  (address-text-style address-type blur? theme)}
                         (if (= address-type :short)
                           (utils/get-short-wallet-address address)
                           address)]]
    (as-> networks $ ;; Take vector of networks
      (into [text/text] network-text-xf $) ;; transform them to colored text and inserts them into [text/text]
      (conj $ address-text)))) ;; Add address-text hiccup at the end
```

Reply:
@ulisesmac thanks a lot for the clarification :)